src/crawlee/sessions/_cookies.py

```diff
@@ -0,0 +1,268 @@
+from __future__ import annotations
+
+from copy import deepcopy
+from http.cookiejar import Cookie, CookieJar
+from typing import TYPE_CHECKING, Any, Literal
+
+from typing_extensions import NotRequired, Required, TypedDict
+
+from crawlee._utils.docs import docs_group
+
+if TYPE_CHECKING:
+    from collections.abc import Iterator
+
+
+@docs_group('Data structures')
+class CookieParam(TypedDict, total=False):
+    """Dictionary representation of cookies for SessionCookies.set() method."""
+
+    name: Required[str]
+    """Cookie name."""
+
+    value: Required[str]
+    """Cookie value."""
+
+    domain: NotRequired[str]
+    """Domain for which the cookie is set."""
+
+    path: NotRequired[str]
+    """Path on the specified domain for which the cookie is set."""
+
+    secure: NotRequired[bool]
+    """Set the `Secure` flag for the cookie."""
```

Comment:
Or does `Secure` reference anything specific?

```suggestion
    """Set the secure flag for the cookie."""
```

Reply:
[Secure](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#secure) - is the exact spelling of the flag in the cookie string.

Reply:
thanks